framework/src/mesh/MooseMesh.C

```diff
@@ -443,6 +444,10 @@ MooseMesh::prepare(const MeshBase * const mesh_to_clone)
     // only subdomain ids are explicitly given
     const auto & add_subdomain_id = getParam<std::vector<SubdomainID>>("add_subdomain_ids");
     _mesh_subdomains.insert(add_subdomain_id.begin(), add_subdomain_id.end());
+
+    // also add the ids to the libmesh subdomain name map
+    for (const auto & id : add_subdomain_id)
+      _mesh->subdomain_name(id) = std::to_string(id);
```

Comment:
I'm not sure if this is the correct thing to do.

Reply:
does it fix the exodus output errors?

Reply:
I don't think it's the right thing to do. The name is supposed to be something meaningful in that map, not just a repeat of the ID. There's no reason to allocate an additional map pair for something like this